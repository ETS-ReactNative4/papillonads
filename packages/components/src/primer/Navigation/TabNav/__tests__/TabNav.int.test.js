import {
  ariaAttr,
  defaultItems as items,
  navigationItems,
  renderCustom,
  renderDefault,
  renderWithButtonActions,
  renderWithTextActions,
  renderWithLinkActions,
  renderWithNavigation,
} from './TabNav.int.render'

jest.mock('@papillonads/library', () => {
  return {
    array: {
      getIndexItems: () => {},
      getIndexItemsWithSelected: () => {},
    },
  }
})

const libraryMockObject = require('@papillonads/library')

describe('<TabNav />', () => {
  const indexItemsDataObject = [
    { ...items[0], index: 0 },
    { ...items[1], index: 1 },
  ]

  const indexNavigationItemsDataObject = [
    { ...navigationItems[0], index: 0 },
    { ...navigationItems[1], index: 1 },
  ]

  const newIndexItemsWithSelectedDataObject = [
    { ...items[0], index: 0, isSelected: true },
    { ...items[1], index: 1, isSelected: false },
  ]

  const newIndexNavigationItemsWithSelectedDataObject = [
    { ...navigationItems[0], index: 0, isSelected: true },
    { ...navigationItems[1], index: 1, isSelected: false },
  ]

  const newItemsDataObject = [
    { ...items[0], isSelected: true },
    { ...items[1], isSelected: false },
  ]

  const newNavigationItemsDataObject = [
    { ...navigationItems[0], isSelected: true },
    { ...navigationItems[1], isSelected: false },
  ]

  const onClickMockFn = jest.fn()

  function mockLibraryForRegularItems() {
    jest.spyOn(libraryMockObject.array, 'getIndexItems').mockReturnValue(indexItemsDataObject)

    jest.spyOn(libraryMockObject.array, 'getIndexItemsWithSelected').mockReturnValue(newIndexItemsWithSelectedDataObject)
  }

  function mockLibraryForNavigationItems() {
    jest.spyOn(libraryMockObject.array, 'getIndexItems').mockReturnValue(indexNavigationItemsDataObject)

    jest.spyOn(libraryMockObject.array, 'getIndexItemsWithSelected').mockReturnValue(newIndexNavigationItemsWithSelectedDataObject)
  }

  afterEach(() => jest.clearAllMocks())

  describe('Event', () => {
    test('must return new items without index when regular items and onClick()', () => {
      mockLibraryForRegularItems()
      const mountCustomRender = global.renderMount(renderCustom(items, onClickMockFn)).find('TabNav')
      mountCustomRender.find('nav').first().find('a').first().simulate('click')
      expect(mountCustomRender.props().onClick).toBe(onClickMockFn)
      expect(mountCustomRender.props().onClick).toHaveBeenCalledWith({
        ariaAttr,
        items: newItemsDataObject,
      })
    })

    test('must return new items without index when navigation items and onClick()', () => {
      mockLibraryForNavigationItems()
      const mountCustomRender = global.renderMount(renderCustom(navigationItems, onClickMockFn)).find('TabNav')
      mountCustomRender.find('nav').first().find('a').first().simulate('click')
      expect(mountCustomRender.props().onClick).toBe(onClickMockFn)
      expect(mountCustomRender.props().onClick).toHaveBeenCalledWith({
        ariaAttr,
        items: newNavigationItemsDataObject,
      })
    })
  })

  describe('Render', () => {
    test('must match renderDefault()', () => {
      mockLibraryForRegularItems()
      expect(global.renderToJSON(renderDefault())).toMatchSnapshot()
    })

    test('must match renderWithButtonActions()', () => {
      mockLibraryForRegularItems()
      expect(global.renderToJSON(renderWithButtonActions())).toMatchSnapshot()
    })

    test('must match renderWithTextActions()', () => {
      mockLibraryForRegularItems()
      expect(global.renderToJSON(renderWithTextActions())).toMatchSnapshot()
    })

    test('must match renderWithLinkActions()', () => {
      mockLibraryForRegularItems()
      expect(global.renderToJSON(renderWithLinkActions())).toMatchSnapshot()
    })

    test('must match renderWithNavigation()', () => {
      mockLibraryForRegularItems()
      expect(global.renderToJSON(renderWithNavigation())).toMatchSnapshot()
    })
  })
})
