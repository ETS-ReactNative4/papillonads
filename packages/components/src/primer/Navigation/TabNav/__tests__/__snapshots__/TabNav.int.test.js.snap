// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabNav /> Render must match renderDefault() 1`] = `
<div
  className="tabnav"
>
  <nav
    aria-label="Foo bar"
    className="tabnav-tabs"
  >
    <a
      aria-current="page"
      className="tabnav-tab"
      href="#url"
      onClick={[Function]}
    >
      Foo tab
    </a>
    <a
      className="tabnav-tab"
      href="#url"
      onClick={[Function]}
    >
      Bar tab
    </a>
  </nav>
</div>
`;

exports[`<TabNav /> Render must match renderWithButtonActions() 1`] = `
<div
  className="tabnav"
>
  <a
    className="btn btn-sm float-right"
    href="#url"
    role="button"
  >
    Button
  </a>
  <nav
    aria-label="Foo bar"
    className="tabnav-tabs"
  >
    <a
      aria-current="page"
      className="tabnav-tab"
      href="#url"
      onClick={[Function]}
    >
      Foo tab
    </a>
    <a
      className="tabnav-tab"
      href="#url"
      onClick={[Function]}
    >
      Bar tab
    </a>
  </nav>
</div>
`;

exports[`<TabNav /> Render must match renderWithLinkActions() 1`] = `
<div
  className="tabnav"
>
  <div
    className="float-right"
  >
    <a
      className="tabnav-extra"
      href="#url"
    >
      Tabnav extra link
    </a>
    <a
      className="tabnav-extra"
      href="#url"
    >
      Tabnav extra link
    </a>
  </div>
  <nav
    aria-label="Foo bar"
    className="tabnav-tabs"
  >
    <a
      aria-current="page"
      className="tabnav-tab"
      href="#url"
      onClick={[Function]}
    >
      Foo tab
    </a>
    <a
      className="tabnav-tab"
      href="#url"
      onClick={[Function]}
    >
      Bar tab
    </a>
  </nav>
</div>
`;

exports[`<TabNav /> Render must match renderWithNavigation() 1`] = `
<div
  className="tabnav"
>
  <nav
    aria-label="Foo bar"
    className="tabnav-tabs"
  >
    <a
      aria-current="page"
      className="tabnav-tab"
      href="#url"
      onClick={[Function]}
    >
      Foo tab
    </a>
    <a
      className="tabnav-tab"
      href="#url"
      onClick={[Function]}
    >
      Bar tab
    </a>
  </nav>
</div>
`;

exports[`<TabNav /> Render must match renderWithTextActions() 1`] = `
<div
  className="tabnav"
>
  <div
    className="tabnav-extra float-right"
  >
    Tabnav widget text here.
  </div>
  <nav
    aria-label="Foo bar"
    className="tabnav-tabs"
  >
    <a
      aria-current="page"
      className="tabnav-tab"
      href="#url"
      onClick={[Function]}
    >
      Foo tab
    </a>
    <a
      className="tabnav-tab"
      href="#url"
      onClick={[Function]}
    >
      Bar tab
    </a>
  </nav>
</div>
`;
